src/main/java/com/example/gradecalculator/repository/GradeTypeRepository.java

```diff
@@ -1,11 +1,11 @@
 package com.example.gradecalculator.repository;
 
-import com.example.gradecalculator.entities.GradeSystem;
 import com.example.gradecalculator.entities.GradeType;
 import org.springframework.data.repository.CrudRepository;
 
-import java.util.List;
+import java.util.Optional;
 
 public interface GradeTypeRepository extends CrudRepository<GradeType, Long> {
-    List<GradeType> findByGradeSystem(GradeSystem gradeSystem);
+   GradeType findByName(String name);
+   Optional<GradeType> findById(Long Id);
```

Comment:
- use Optional of expected type when using find methods unless you are sure there is always an result that is not null